examples/VLCB_4in4out/VLCB_4in4out.ino

```diff
@@ -219,64 +223,89 @@ void loop()
   }
 
   // test for switch input
-  processSwitches();
+  check = false;
+  processSwitches(check);
 
   // bottom of loop()
 }
 
-void processSwitches(void) {
-  bool isSuccess = true;
-  for (byte i = 0; i < NUM_SWITCHES; i++) {
-    moduleSwitch[i].update();
-    if (moduleSwitch[i].changed()) {      
-      byte nv = i+1;
-      byte nvval = modconfig.readNV(nv);
-      bool state;
-
-      //DEBUG_PRINT(F("> Button ") << i << F(" state change detected"));
-      //DEBUG_PRINT(F(" NV Value = ") << nvval);
-
-      switch (nvval) {
-        case 0:
-          // ON and OFF
-          state = (moduleSwitch[i].fell());
-          //DEBUG_PRINT(F("> Button ") << i << (moduleSwitch[i].fell() ? F(" pressed, send state: ") : F(" released, send state: ")) << state);
-          epService.sendEvent(state, i);
-          break;
-
-        case 1:
-          // Only ON
-          if (moduleSwitch[i].fell()) {
-            state = true;
-            //DEBUG_PRINT(F("> Button ") << i << F(" pressed, send state: ") << state);
-            epService.sendEvent(state, i);
-          }
-          break;
-
-        case 2:
-          // Only OFF
-          if (moduleSwitch[i].fell()) {
-            state = false;
-            //DEBUG_PRINT(F("> Button ") << i << F(" pressed, send state: ") << state);
-            epService.sendEvent(state, i);
-          }
-          break;
-
-        case 3:
-          // Toggle button
-          if (moduleSwitch[i].fell()) {
-            switchState[i] = !switchState[i];
-            state = (switchState[i]);
-            //DEBUG_PRINT(F("> Button ") << i << (moduleSwitch[i].fell() ? F(" pressed, send state: ") : F(" released, send state: ")) << state);
+byte checkProduced(void)
+{
+  check = true;
+  byte pIndex = processSwitches(check);
+  return pIndex;
+}
+
+byte processSwitches(bool check) {
+    
+  if (check)
+  {
+    for (byte i = 0; i < NUM_SWITCHES; i++) 
+    {
+      moduleSwitch[i].update();      
+      if (moduleSwitch[i].changed())
+      {
+        //DEBUG_PRINT(F("sk> Check index is ") << i);
+        return i;
```

Comment:
You don't check the NVs here for what is happening on each input happening. Say that you only have an event on closing a switch. This code will also trigger on breaking the switch.